### ISSUE TITLE
"Legacy Island" Foliage should be included in Gridmap

### ISSUE THREAD

**Issue by TheYellowArchitect** [foxssake/netfox#180]
idk why FPS is so low in forest brawl without even connecting. Anyway, for a simpler demo scene, foliage should be included in the gridmap for batch rendering
![gridmap-no-foliage](https://github.com/foxssake/netfox/assets/25630803/397c5a04-3e26-4f1e-88b5-ab0c33efd398)

Same as "Three Peaks" basically

**Comment by elementbound** [comment]
Hi, thanks for checking the project! Could you please give some more details? E.g.:
- What's your current FPS
- What's the FPS you'd like to see
- What's your hardware setup
- Any render profiling data - I think we can be most efficient if we can identify the performance bottleneck, if there's any :slightly_smiling_face: 

Also fyi the legacy islands map is not used currently in the game, is there some specific use case for you where it appears?

**Comment by TheYellowArchitect** [comment]
>What's your current FPS
What's the FPS you'd like to see
What's your hardware setup

Depends on how many instances I run, but with 1 instance, on "Three Peaks" I get 240 FPS (which I learned is capped, but would be a red flag for a naive developer)
With 4 instances, I get 101, 151, 153 on the 3 non-focused instances, and **53** on the focused instance. I don't have a low-end pc...
Note this is without pressing anything, havent clicked any UI button.

>What's your hardware setup

Godot v4.2.stable - Artix Linux #1 SMP PREEMPT_DYNAMIC Wed, 03 Jan 2024 18:30:58 +0000 - Tty - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1050 Ti (nvidia; 545.29.06) - AMD Ryzen 5 2600 Six-Core Processor (12 Threads)

>Any render profiling data - I think we can be most efficient if we can identify the performance bottleneck, if there's any 🙂

I uploaded a clip of the whole setup example

https://github.com/foxssake/netfox/assets/25630803/e1ce19f1-c143-4732-bb04-5492a7efd1a1

>Also fyi the legacy islands map is not used currently in the game, is there some specific use case for you where it appears?

Nope, was just scouting the files, and found it weird that flowers have gridmap objects but exist as seperate objects. Having not clicked the three peaks scene, I thought legacy islands map was what loaded and tried to optimize it :sweat_smile:

**Comment by TheYellowArchitect** [comment]
I would do a pull request for this, but the project is in 4.1 but I use 4.3-dev2 so there are a loooooot more changes which I would rather not filter in the git add

I suggest updating the project when Godot 4.3 comes out, or when these commits are merged:
https://github.com/godotengine/godot/pull/87190 https://github.com/godotengine/godot/pull/87186 https://github.com/godotengine/godot/pull/87185

**Comment by elementbound** [comment]
There's 10 foliage objects in a map that is not used and is mostly in the repo for historic reasons. The idea behind not having them in the gridmap was to be able to freely place objects easier. If 10 objects are a performance issue, then there's much more to worry about.

In Three peaks, all the foliage is included in gridmap. 

I'm qualifying the original scope of the issue as wontfix as it relates to an unused scene, and is not a performance bottleneck.

> With 4 instances, I get 101, 151, 153 on the 3 non-focused instances, and 53 on the focused instance. 

Yes, that 53 does sound a bit fishy. The issue is it's really hard to debug these, especially when they occur on somebody else's PC. It's also difficult to determine if this is caused by some background process of the OS, some background process of the engine, the GPU encountering something unusual, game code doing something unexpected, or the addon code itself. Also not sure how running the same game 4 times in parallel affects performance characteristics.

Once #173 is done, we should have more certainty about the addon code performing as expected.

> I would do a pull request for this, but the project is in 4.1 but I use 4.3-dev2

If you find yourself wanting to use multiple Godot versions, I can recommend [Godot Manager](https://github.com/eumario/godot-manager) :slightly_smiling_face: For the version upgrade - although I don't have an official policy at the moment - I aim to support as many 4.x versions as I reasonably can, for which staying on 4.1 is the most practical way currently.

tl;dr: Original scope shouldn't affect perf so closing as *wontfix*, but some valid points were raised in the discussion.
